test/unit/test_retry_network.py

```diff
@@ -412,3 +426,53 @@ def error_recv_into(*args, **kwargs):
             not in caplog.text
         )
         assert caplog.text.count("Starting new HTTPS connection") > 1
+
+
+@pytest.mark.parametrize("next_action", ("RETRY", "ERROR"))
+@patch("snowflake.connector.vendored.requests.sessions.Session.request")
+def test_login_request_timeout(mockSessionRequest, next_action):
+    """For login requests, all errors should be bubbled up as OperationalError for authenticator to handle"""
+    mockSessionRequest.side_effect = mock_request_with_action(next_action)
+
+    connection = mock_connection()
+    rest = SnowflakeRestful(
+        host="testaccount.snowflakecomputing.com", port=443, connection=connection
+    )
+
+    with pytest.raises(OperationalError):
+        rest.fetch(
+            method="post",
+            full_url="https://testaccount.snowflakecomputing.com/session/v1/login-request",
+            headers=dict(),
+        )
+
+
+@pytest.mark.flaky(reruns=3)
```

Comment:
Why is a unit test flaky? Could you make it not flaky, please?

Reply:
I made it flaky because very occasionally, a test on the server side would fail but succeed on retry. Maybe because of thread scheduling? In hindsight though, I see why we wouldn't want to mark unit tests like these flaky.